src/viadot/sources/duckdb.py

```diff
@@ -0,0 +1,249 @@
+            df = self.run_query(query, fetch_type="dataframe")
+            if df.empty:
+                self._handle_if_empty(if_empty=if_empty)
+        else:
+            df = pd.DataFrame()
+        return df
+
+    def run_query(
+        self, query: str, fetch_type: Literal["record", "dataframe"] = "record"
+    ) -> Union[List[Record], bool]:
+        """Run a query on DuckDB.
+
+        Args:
+            query (str): The query to execute.
+            fetch_type (Literal[, optional): How to return the data: either
+            in the default record format or as a pandas DataFrame. Defaults to "record".
+
+        Returns:
+            Union[List[Record], bool]: Either the result set of a query or,
+            in case of DDL/DML queries, a boolean describing whether
+            the query was excuted successfuly.
+        """
+        allowed_fetch_type_values = ["record", "dataframe"]
+        if fetch_type not in allowed_fetch_type_values:
+            raise ValueError(
+                f"Only the values {allowed_fetch_type_values} are allowed for 'fetch_type'"
+            )
+        cursor = self.con.cursor()
+        cursor.execute(query)
+
+        query_clean = query.upper().strip()
+        regex = r"^\s*[--;].*"
+        lines = query_clean.splitlines()
+        final_query = ""
+
+        for line in lines:
+            line = line.strip()
+            match_object = re.match(regex, line)
+            if not match_object:
+                final_query += " " + line
+        final_query = final_query.strip()
+        query_keywords = ["SELECT", "SHOW", "PRAGMA", "WITH"]
+        if any(final_query.startswith(word) for word in query_keywords):
+            if fetch_type == "record":
+                result = cursor.fetchall()
+            else:
+                result = cursor.fetchdf()
+        else:
+            result = True
+
+        cursor.close()
+        return result
+
+    def _handle_if_empty(self, if_empty: str = "warn") -> NoReturn:
+        if if_empty == "warn":
+            self.logger.warning("The query produced no data.")
+        elif if_empty == "skip":
+            raise SKIP("The query produced no data. Skipping...")
+        elif if_empty == "fail":
+            raise ValueError("The query produced no data.")
+
+    def create_table_from_parquet(
+        self,
+        table: str,
+        path: str,
+        schema: str = None,
+        if_exists: Literal["fail", "replace", "append", "skip", "delete"] = "fail",
+    ) -> NoReturn:
+        """Create a DuckDB table with a CTAS from Parquet file(s).
+
+        Args:
+            table (str): Destination table.
+            path (str): The path to the source Parquet file(s). Glob expressions are
+            also allowed here (eg. `my_folder/*.parquet`).
+            schema (str, optional): Destination schema. Defaults to None.
+            if_exists (Literal[, optional): What to do if the table already exists. Defaults to "fail".
```

Comment:
Most of them are self explanatory, but it would be useful to document the difference between "replace" and "delete" in the docstring.